JoinColony/colonyDapp
# Improve Transfer NFT loading pattern and tidy other sections

**Note: only the first commit belongs uniquely to this PR. Others from merge of other pr.**
## Description

This PR refactors the Transfer NFT's loading flow  (inspired by doing the same at #3979). It also implements a few other tweaks to make the transaction sections more consistent, such as updating message names and reusing common components.
 
**Changes** 🏗

* Extracting common code into `shared.ts` file (1) and refactoring Transfer NFT loading flow (1)

## To test

1. Just double check that in the case of Transfer NFT, the loading spinner only shows up the first time we fetch the NFTs for the safe and that there's no ui flashing when, e.g. clicking "back" from the Transaction Preview.
